packages/web/src/pages/dashboard-page/components/hooks.ts

```diff
@@ -0,0 +1,368 @@
+import { useMemo } from 'react'
+
+import {
+  Collection,
+  Track,
+  isContentCollectibleGated,
+  isContentFollowGated,
+  isContentTipGated,
+  isContentUSDCPurchaseGated
+} from '@audius/common/models'
+import { accountSelectors } from '@audius/common/store'
+import {
+  IconCart,
+  IconCollectible,
+  IconSpecialAccess,
+  IconVisibilityHidden,
+  IconVisibilityPublic
+} from '@audius/harmony'
+import { useSelector } from 'react-redux'
+import { Nullable } from 'vitest'
+
+import { makeGetDashboard } from '../store/selectors'
+
+import {
+  AlbumFilters,
+  DataSourceAlbum,
+  DataSourceTrack,
+  TrackFilters
+} from './types'
+
+const { getAccountAlbums } = accountSelectors
+
+const messages = {
+  public: 'Public',
+  premium: 'Premium',
+  specialAcess: 'SpecialAccess',
+  gated: 'Gated',
+  hidden: 'Hidden'
+}
+
+/** ------------------------ Tracks ------------------------ */
+
+const formatTrackMetadata = (metadata: Track, i: number): DataSourceTrack => {
+  return {
+    ...metadata,
+    key: `${metadata.title}_${metadata.dateListened}_${i}`,
+    name: metadata.title,
+    date: metadata.created_at,
+    time: metadata.duration,
+    saves: metadata.save_count,
+    reposts: metadata.repost_count,
+    plays: metadata.play_count
+  }
+}
+
+export const useFormattedTrackData = () => {
+  const { tracks } = useSelector(makeGetDashboard())
+  return tracks
```

Comment:
Depending on how you use it, this is going to change every time it runs. Might be worth seeing how often it's recomputing and maybe adding a memo based on the value of `tracks`